app/cdap/components/Replicator/Create/Content/SelectColumnsWithTransforms/index.tsx

```diff
@@ -113,36 +115,69 @@ const SelectColumnsView = (props: ISelectColumnsProps) => {
       body.schema = props.tableInfo.schema;
     }
 
-    Observable.forkJoin([
-      MyReplicatorApi.getTableInfo(params, body),
-      MyReplicatorApi.assessTable(params, body),
-    ]).subscribe({
-      next: (value: any) => {
-        const [tableInfo, assessedTable] = value;
-        const combinedColumns = tableInfo.columns.map((t1) => ({
-          ...t1,
-          ...assessedTable.columns.find((t2) => t2.sourceName === t1.name),
-        }));
-        const selectedColumns = getInitialSelectedColumns(combinedColumns);
-        dispatch({
-          type: 'setTableInfo',
-          payload: {
-            columns: combinedColumns,
-            primaryKeys: tableInfo.primaryKey,
-            selectedColumns,
-            filteredColumns: combinedColumns,
-            selectedReplication:
-              selectedColumns.size === 0 ? ReplicateSelect.all : ReplicateSelect.individual,
-          },
-        });
-      },
-      error: (err) => {
-        dispatch({ type: 'setError', payload: err });
-      },
-      complete: () => {
-        dispatch({ type: 'setLoading', payload: false });
-      },
+    // need to check if the current table is selected
+    const tableKey = generateTableKey(body);
+    let needUpdateTables = true;
+    props.tables.forEach((table) => {
+      if (generateTableKey(table) === tableKey) {
+        // already selected, no need to update selected tables
+        needUpdateTables = false;
+        return;
+      }
     });
+    if (needUpdateTables) {
+      let updatedSelectedTables = props.tables;
```

Comment:
Do we want to mutate the `tables` object? If not, perhaps we need to use the spread operator?
`let updatedSelectedTables = {...props.tables};`